src/Twig/Extension/PageExtension.php

```diff
@@ -267,13 +253,13 @@ public function renderBlock(PageBlockInterface $block, array $options = [])
      *
      * @return ControllerReference
      */
-    public function controller($controller, $attributes = [], $query = [])
+    public function controller(Environment $env, $controller, $attributes = [], $query = [])
```

Comment:
why don't you inject this in `__construct`?

Reply:
because https://www.lrotherfield.com/blog/symfony2-get-twig-environment-twig-extension

but what i would inject in construct is RequestStack

Reply:
Well I would inject this in constructor and not in the method, because it's a dependency that your class needs to work, then IMO it should be injected on constructor

Reply:
https://symfony.com/doc/current/templates.html#rendering-a-template-in-services

Reply:
and could you use `$twig` instead of `$env`, like symfony example, it is quite clear that you are inject twig class

Reply:
the problem is that this is a twig extension, not some general service

defining Twig Env in Constructor causes circular dependency problems

Reply:
Hmmmm